src/rebuild/rebuild_iv.c

```diff
@@ -210,35 +210,54 @@ rebuild_iv_ent_refresh(struct ds_iv_entry *entry, struct ds_iv_key *key,
 		 * node changed.
 		 */
 		rc = crt_group_rank(NULL, &rank);
-		if (dst_iv->riv_global_done && rc == 0 &&
-		    d_rank_in_rank_list(rpt->rt_svc_list, rank)) {
+		if (dst_iv->riv_global_done && rc == 0) {
 			struct daos_rebuild_status rs = { 0 };
-
-			rs.rs_version	= src_iv->riv_ver;
-			rs.rs_errno	= src_iv->riv_status;
-			rs.rs_done	= 1;
-			rs.rs_obj_nr	= src_iv->riv_obj_count;
-			rs.rs_rec_nr	= src_iv->riv_rec_count;
-			rs.rs_toberb_obj_nr	=
-				src_iv->riv_toberb_obj_count;
-			rs.rs_size	= src_iv->riv_size;
-
-			rc = rebuild_status_completed_update(
-					src_iv->riv_pool_uuid, &rs);
-			if (rc != 0) {
-				D_ERROR("_status_completed_update, "
-					DF_UUID" failed, rc %d.\n",
-					DP_UUID(src_iv->riv_pool_uuid),
-					rc);
+			daos_prop_t	*prop = NULL;
+			struct daos_prop_entry *prop_entry;
+			d_rank_list_t	*svc_list;
+
+			D_ALLOC_PTR(prop);
+			if (prop == NULL)
+				D_GOTO(out, rc = -DER_NOMEM);
+
+			rc = pool_iv_prop_fetch(rpt->rt_pool, prop);
+			if (rc)
+				D_GOTO(free, rc);
+
+			prop_entry = daos_prop_entry_get(prop,
+						    DAOS_PROP_PO_SVC_LIST);
```

Comment:
is it only to check the rank in svc_list? maybe the gain is less than lose as the rebuild_status_completed_update() is very cheap, and not very sure that if adding the iv fetch possibly involve failure if with node failure in IV path.

Reply:
oh, we also use svc_list to open the container for fetching/iterating the objects from remote server.